In [1]:
import numpy as np
A = np.array([[56.0,0.0,4.4,68.0],
             [1.2,104.0,52.0,8.0],
             [1.8,135.0,99.0,0.9]])
print A

[[  56.     0.     4.4   68. ]
 [   1.2  104.    52.     8. ]
 [   1.8  135.    99.     0.9]]


In [2]:
cal = A.sum(axis=0)

In [3]:
print cal

[  59.   239.   155.4   76.9]


In [4]:
percentage = 100*A/cal

In [5]:
print percentage

[[ 94.91525424   0.           2.83140283  88.42652796]
 [  2.03389831  43.51464435  33.46203346  10.40312094]
 [  3.05084746  56.48535565  63.70656371   1.17035111]]


## 3x4 3行4列的矩阵 与 1x4 1行4列的矩阵运算 会先将1x4的矩阵 复制2行成3x4的矩阵 再做运算

## 求每个元素的比值 94% 2.03%

In [6]:
B = np.array([[1,2]])
print B

[[1 2]]


In [7]:
C = np.array([[0.1],[0.1]])
print C

[[ 0.1]
 [ 0.1]]


In [8]:
print B+C

[[ 1.1  2.1]
 [ 1.1  2.1]]


## B 1行2列  C 2行一列 复制一列成 2行2列    B 复制一行成2行2列 相加 得到 2行2列

## a = np.array([1,2,3,4]) a为列表or numpy数组
-----
## a = np.array([[1,2,3,4]]) a为向量

# 注意事项

In [9]:
b = np.random.randn(5) #b 为 长度为5的numpy数组
print b

[ 0.7118177  -0.14645218 -0.67843133  1.38068008  1.82062897]


In [11]:
print b.shape,type(b) # b.reshape(5,1)转换为5x1的向量 矩阵

(5L,) <type 'numpy.ndarray'>


In [12]:
b = np.random.randn(5,1) #b 为5x1的向量 矩阵
print b

[[-0.76948072]
 [-2.28616048]
 [-0.94598391]
 [-0.3737337 ]
 [-0.22999546]]


In [13]:
print b.T

[[-0.76948072 -2.28616048 -0.94598391 -0.3737337  -0.22999546]]


In [14]:
print np.dot(b,b.T)

[[ 0.59210058  1.75915641  0.72791638  0.28758088  0.17697707]
 [ 1.75915641  5.22652976  2.16267105  0.85441523  0.52580653]
 [ 0.72791638  2.16267105  0.89488557  0.35354607  0.217572  ]
 [ 0.28758088  0.85441523  0.35354607  0.13967688  0.08595705]
 [ 0.17697707  0.52580653  0.217572    0.08595705  0.05289791]]


In [16]:
assert(b.shape == (6,1))  # assert声明 检查维度，b是否是6x1的向量矩阵

AssertionError: 